### ISSUE TITLE
FunctionCompletion - end-char isn't handled correctly plus some suggestions

### ISSUE THREAD

**Issue by siggemannen** [bobbylight/AutoComplete#69]
Hi,
I noticed that if you don't want to add a "end" character for your function call, you can't due to:

```
sb.append(getProvider().getParameterListEnd());
```
it should do something like this:
```
char paramListEnd = getProvider().getParameterListEnd();
if (paramListEnd !='\0') {
			sb.append(paramListEnd);
		}
```

Also, there's no easy way to override this class, due to the fact that ParameterizedCompletionInsertionInfo is package private. Perhaps ParameterizedCompletionInsertionInfo should be made public?

**Comment by bobbylight** [comment]
Better late than never!  This seems to have been addressed in commit cc3742aa on 2021-06-10 😛

**Comment by bobbylight** [comment]
Marking as the 3.1.5 release even though tis was addressed in an earlier release, just so it's doc'd as completed.
